standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -10548,13 +10565,20 @@ public void drop_stored_procedure(StoredProcedureRequest request) throws MetaExc
     }
   }
 
-public Package find_package(GetPackageRequest request) throws MetaException {
+public Package find_package(GetPackageRequest request) throws MetaException, NoSuchObjectException {
     startFunction("find_package");
     Exception ex = null;
     try {
-      return getMS().findPackage(request);
+      Package pkg = getMS().findPackage(request);
+      if (pkg == null) {
+        throw new NoSuchObjectException(
+                "HPL/SQL package " + request.getDbName() + "." + request.getPackageName() + " does not exist");
+      }
+      return pkg;
     } catch (Exception e) {
-      LOG.error("Caught exception", e);
+      if (!(e instanceof NoSuchObjectException)) {
+        LOG.error("Caught exception", e);
```

Comment:
We already throw the exception. Why do we need to log it here? Are exceptions thrown from here swallowed somewhere up the call stack?

Reply:
No, but this is happening on HMS side. We log the exception in the client side. The former is written to metastore logs  while the latter in hs2 logs and the stacktraces are not the same. Many of the other methods of hms handler do this as well.